### ISSUE TITLE
Project can migrate to apply-refact, hlint --refactor use

### ISSUE THREAD

**Issue by Anton-Latukha** [hvr/Hs2010To201x#2]
They can transform the body of source code.

In the face of future soft deprecations, this tool I think may get another wave of use.

**Comment by emberdart** [comment]
I see that using hlint this can be converted using lhs / rhs in just using `pure` outside of instance definitions but what about instance definitions?

**Comment by Anton-Latukha** [comment]
Yes. It is a part of the solution, `hlint` can not be a full solution, but I lack in depth `apply-refact` knowledge to know its superset of possibilities compared to what is available in `hlint` rules.

`apply-refact` also *afaik* still has that annoying bug when it can not handle the CPP and deletes it.
